chain-network/src/grpc/convert.rs

```diff
@@ -173,97 +156,18 @@ impl IntoProtobuf for Gossip {
     }
 }
 
-impl FromProtobuf<proto::Peer> for Peer {
-    fn from_message(message: proto::Peer) -> Result<Self, Error> {
-        use proto::peer;
-
-        match message.peer {
-            Some(peer::Peer::V4(pv4)) => {
-                let port = pv4.port as u16;
-                let segments = pv4.ip.to_be_bytes();
-                let ipv4 = Ipv4Addr::new(segments[0], segments[1], segments[2], segments[3]);
-                let addr = SocketAddr::new(IpAddr::V4(ipv4), port);
-                Ok(addr.into())
-            }
-            Some(peer::Peer::V6(pv6)) => {
-                let port = pv6.port as u16;
-                let ipv6 = unserialize_ipv6(pv6.ip_high, pv6.ip_low);
-                let addr = SocketAddr::new(IpAddr::V6(ipv6), port);
-                Ok(addr.into())
-            }
-            None => Err(Error::new(
-                error::Code::InvalidArgument,
-                "invalid Peer payload, one of the fields is required",
-            )),
-        }
-    }
-}
-
-fn unserialize_ipv6(high: u64, low: u64) -> Ipv6Addr {
-    let h = high.to_be_bytes();
-    let l = low.to_be_bytes();
-    fn from_be_bytes(h: u8, l: u8) -> u16 {
-        u16::from_be_bytes([h, l])
-    }
-    let segments: [u16; 8] = [
-        from_be_bytes(h[0], h[1]),
-        from_be_bytes(h[2], h[3]),
-        from_be_bytes(h[4], h[5]),
-        from_be_bytes(h[6], h[7]),
-        from_be_bytes(l[0], l[1]),
-        from_be_bytes(l[2], l[3]),
-        from_be_bytes(l[4], l[5]),
-        from_be_bytes(l[6], l[7]),
-    ];
-    std::net::Ipv6Addr::new(
-        segments[0],
-        segments[1],
-        segments[2],
-        segments[3],
-        segments[4],
-        segments[5],
-        segments[6],
-        segments[7],
-    )
-}
-
-impl IntoProtobuf for Peer {
-    type Message = proto::Peer;
-
-    fn into_message(self) -> proto::Peer {
-        let peer = match self.addr() {
-            SocketAddr::V4(v4addr) => {
-                let port: u32 = v4addr.port().into();
-                let ip = u32::from_be_bytes(v4addr.ip().octets());
-                proto::peer::Peer::V4(proto::PeerV4 { ip, port })
-            }
-            SocketAddr::V6(v6addr) => {
-                let port: u32 = v6addr.port().into();
-                let (ip_high, ip_low) = serialize_ipv6(v6addr.ip());
-                proto::peer::Peer::V6(proto::PeerV6 {
-                    port,
-                    ip_high,
-                    ip_low,
-                })
-            }
+impl FromProtobuf<proto::PeersResponse> for Gossip {
+    fn from_message(message: proto::PeersResponse) -> Result<Self, Error> {
+        let gossip = Gossip {
+            nodes: message
+                .peers
+                .into_iter()
+                .map(Node::from_bytes)
```

Comment:
Is there `try_from_bytes`? It would make much more sense since this is the input from the outside world (we are not sure it valid) and the calling function is already fallible.

Reply:
well, `Node` is just a `Box<[u8]>` in this context, I guess we can't really enforce properties at this level

Reply:
Ah, well, I did not look at the type definition.